src/core/components/hv-route/index.tsx

```diff
@@ -438,6 +430,17 @@ export default function HvRoute(props: Types.Props) {
     routeDocContext,
   );
 
+  React.useEffect(() => {
+    if (props.navigation) {
+      const unsubscribe = props.navigation.addListener('focus', () => {
+        NavigatorService.setSelected(routeDocContext, props.route?.params?.id);
+      });
+
+      return unsubscribe;
```

Comment:
I'm not too familiar with the useEffect API, but does it automatically remove the listener on unmount?